tests/python/gpu/test_fusion.py

```diff
@@ -0,0 +1,195 @@
+    for op_name in unary_ops:
+        print("Checking fusion of " + op_name)
+        op = getattr(mx.sym, op_name)
+        sym = op(a)
+        check_fused_symbol(sym, a=arr)
+
+    # unary ops requiring special treatment
+
+    # arccosh needs input to be >= 1
+    arr2 = arr + 1
+    check_fused_symbol(mx.sym.arccosh(a), a=arr2)
+
+    # Activation requires act_type attribute
+    for act_type in ['relu', 'sigmoid', 'tanh', 'softrelu', 'softsign']:
+        check_fused_symbol(mx.sym.Activation(a, act_type=act_type), a=arr)
+
+    # Cast requires dtype
+    for dtype in ['float16', 'float32', 'float64', 'int32']:
+        check_fused_symbol(mx.sym.Cast(a, dtype=dtype), a=arr)
+
+    # reshape requires shape
+    check_fused_symbol(mx.sym.reshape(a, shape=(-1,)), a=arr)
+
+    # expand_dims requires axis
+    check_fused_symbol(mx.sym.expand_dims(a, axis=1), a=arr)
+
+    # clip requires a_min, a_max
+    check_fused_symbol(mx.sym.clip(a, a_min=0.3, a_max=0.7), a=arr)
+
+    # smooth_l1 requires a scalar
+    check_fused_symbol(mx.sym.smooth_l1(a, scalar=0.3), a=arr)
+
+def check_binary_ops():
+    a = mx.sym.Variable('a')
+    b = mx.sym.Variable('b')
+    shape = rand_shape_2d()
+    arr1 = mx.random.uniform(shape=shape)
+    arr2 = mx.random.uniform(shape=shape)
+
+    check_fused_symbol(a+b, a=arr1, b=arr2)
+    check_fused_symbol(a+3, a=arr1)
+    check_fused_symbol(a-b, a=arr1, b=arr2)
+    check_fused_symbol(a-3, a=arr1)
+    check_fused_symbol(3-a, a=arr1)
+    check_fused_symbol(a*b, a=arr1, b=arr2)
+    check_fused_symbol(a*3, a=arr1)
+    check_fused_symbol(a/b, a=arr1, b=arr2)
+    check_fused_symbol(a/3, a=arr1)
+    check_fused_symbol(3/a, a=arr1)
+    check_fused_symbol(a**b, a=arr1, b=arr2)
+    check_fused_symbol(a**3, a=arr1)
+    check_fused_symbol(mx.sym.pow(3,a), a=arr1)
+    check_fused_symbol(mx.sym.maximum(a,b), a=arr1, b=arr2)
+    check_fused_symbol(mx.sym.minimum(a,b), a=arr1, b=arr2)
+    check_fused_symbol(mx.sym.hypot(a,b), a=arr1, b=arr2)
+    check_fused_symbol(mx.sym.hypot(a,3), a=arr1)
+
+def check_other_ops():
+    a = mx.sym.Variable('a')
+    b = mx.sym.Variable('b')
+    c = mx.sym.Variable('c')
+    shape = rand_shape_2d()
+    shape = (5,) + shape
+    arr1 = mx.random.uniform(shape=shape)
+    arr2 = mx.random.uniform(shape=shape)
+    arr3 = mx.random.uniform(shape=shape)
+
+    check_fused_symbol(mx.sym.add_n(a,b,c), a=arr1, b=arr2, c=arr3)
+
+    check_fused_symbol(mx.sym.slice_axis(a, axis=0, begin=1, end=4), a=arr1)
+
+    begin = (random.randint(0, shape[0]-2),
+             random.randint(0, shape[1]-2),
+             random.randint(0, shape[2]-2))
+    end = (random.randint(begin[0]+1, shape[0]-1),
+           random.randint(begin[1]+1, shape[1]-1),
+           random.randint(begin[2]+1, shape[2]-1))
+    check_fused_symbol(mx.sym.slice(a, begin=begin, end=end), a=arr1)
+
+@with_seed()
+def test_fusion():
+    check_unary_ops()
+    check_binary_ops()
+    check_other_ops()
```

Comment:
Nosetests does not need main block, it is more of a convenience so that you are able to run the test script via `python test_fusion.py` instead of `nosetests test_fusion.py`. That said, most tests have it, so I will add it here as well.